shared/oae/js/jquery-plugins/jquery.switchtab.js

```diff
@@ -0,0 +1,164 @@
+ * BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+/**
+ * Utility plugin that handles the "switch tab" component interactions.
+ */
+
+define(['jquery', 'jquery.history'], function (jQuery, oaeUtil) {
+    (function($) {
+
+        /**
+         * OAE plugin that implements the interaction of the "Switch Tab" component. A Switch Tab
+         * is essentially a set of tabs that look like a left-to-right switch. There can only be
+         * one selected at one time. Back functionality is also provided.
+         *
+         * Usage:
+         *
+         *  You should create a template that looks like the following:
+         *
+         *  ```
+         *  <div id="search-scope" class="oae-switchtab-container">
+         *      <div class="oae-switchtab-control">
+         *          <button class="oae-switchtab-switch" data-switchtab-id="all" data-switchtab-path="/search/all">
+         *              <span class="oae-switchtab-switch-icon fa fa-globe"></span>
+         *              <span class="oae-switchtab-switch-label">__MSG__EVERYWHERE__</span>
+         *          </button>
+         *          <button class="oae-switchtab-switch" data-switchtab-id="my" data-switchtab-path="/search/my">
+         *              <span class="oae-switchtab-switch-icon fa fa-lock"></span>
+         *              <span class="oae-switchtab-switch-label">__MSG__MY_STUFF_ONLY__</span>
+         *          </button>
+         *      </div>
+         *  </div>
+         *  ```
+         *
+         * Where all classes prefixed with `oae-switchtab` are specific to the switchtab component.
+         * The following special data attributes are available:
+         *
+         *  *   data-switchtab-id (required): Indicates that id of the tab. It is used both
+         *      internally to indentify the element and it is useful for the $.switchtabId() method
+         *      in order to determine which tab is currently selected
+         *  *   data-switchtab-path (required): Indicates not only at what path prefix the tab
+         *      should be active, but also what path to push into the History module when the tab
+         *      is clicked. Note that when navigating to a tab, the query string is retained
+         *  *   data-switchtab-title (optional): If specified, indicates what browser title should
+         *      be used for the tab
+         *
+         * To perform an action when the tab is changed, you should listen to `$(window).on('statechange')`
+         * and use `$mySwitchTab.switchtabId()` to get the tab id of the currently active tab.
+         */
+        $.fn.switchtab = function() {
+            var $switchtab = $(this);
+
+            var currentId = _idFromState($switchtab);
+
+            $switchtab.on('click', '.oae-switchtab-switch', function() {
```

Comment:
I don't think using bootstrap tabs JS interaction is the right way to go as they have disabled url changing / back support. I've implemented their HTML template and overridden their styles, but I think that's about as far as we can go.